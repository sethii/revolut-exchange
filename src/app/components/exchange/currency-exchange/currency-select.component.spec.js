import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';
import { mount } from 'enzyme';

import { CurrencySelect } from './currency-select.component';

describe('Currency select component', () => {
  it('renders currency select', () => {
    const onCurrencySelect = jest.fn();
    const currencies = [
      {
        code: 'USD',
      },
      {
        code: 'PLN'
      }
    ];
    const selectedCurrency = {
      code: 'PLN'
    };
    const CurrencySelectComponent = toClass(CurrencySelect);

    expect(renderer.create(<CurrencySelectComponent
      currencies={currencies}
      selectedCurrency={selectedCurrency}
      onCurrencyChange={onCurrencySelect}
    />).toJSON()).toMatchSnapshot();
  });

  it('it changes selected currency', () => {
    const onCurrencySelect = jest.fn();
    const currencies = [
      {
        code: 'USD',
      },
      {
        code: 'PLN'
      }
    ];
    const selectedCurrency = {
      code: 'PLN'
    };
    const CurrencySelectComponent = toClass(CurrencySelect);
    const wrapper = mount(<CurrencySelectComponent
      currencies={currencies}
      selectedCurrency={selectedCurrency}
      onCurrencyChange={onCurrencySelect}
    />);

    wrapper.find('select').simulate('change', { target: { value: 0 } });

    expect(onCurrencySelect).toBeCalledWith(currencies[0]);
  });
});
