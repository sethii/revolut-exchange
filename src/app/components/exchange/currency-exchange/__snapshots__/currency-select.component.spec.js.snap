// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency select component renders currency select 1`] = `
.glamor-1 {
  height: 43px;
  width: 80px;
  outline: none;
}

<select
  className="glamor-0 glamor-1"
  onChange={[Function]}
  value={1}
>
  <option
    value={0}
  >
    USD
  </option>
  <option
    value={1}
  >
    PLN
  </option>
</select>
`;
